src/nodes/Table.js

```diff
@@ -0,0 +1,110 @@
+import { Table } from '@tiptap/extension-table'
+import { Node, mergeAttributes } from '@tiptap/core'
+import { TextSelection } from 'prosemirror-state'
+
+/*
+ * Markdown tables do not include captions.
+ * We still need to parse them though
+ * because otherwise tiptap will try to insert their text content
+ * and put it in the top row of the table.
+ */
+const tableCaption = Node.create({
+	name: 'tableCaption',
+	content: 'inline*',
+	addAttributes() {
+		return {}
+	},
+
+	renderHTML() {
+		return ['caption']
+	},
+
+	toMarkdown(state, node) {
+	},
+
+	parseHTML() {
+		return [
+			{ tag: 'table caption', priority: 90 },
+		]
+	},
+
+})
+
+function getTableNodeTypes(schema) {
+	if (schema.cached.tableNodeTypes) {
+		return schema.cached.tableNodeTypes
+	}
+
+	const roles = {}
+
+	Object.keys(schema.nodes).forEach(type => {
+		const nodeType = schema.nodes[type]
+
+		if (nodeType.spec.tableRole) {
+			roles[nodeType.spec.tableRole] = nodeType
+		}
+	})
+
+	schema.cached.tableNodeTypes = roles
+
+	return roles
+}
+
```

Comment:
Asked for exporting this in https://github.com/ueberdosis/tiptap/issues/2621

Reply:
Stopped using this as `tableRoles` are used multiple times in our implementation now. We need the row type for the first line that only includes `<th>` and the other rows as well. So identifying node types by table role is no longer a good idea.